### PR TITLE
CXXCBC-52: implement replica read operations

### PR DESCRIPTION
**Pull request by avsej**

### CODE REVIEW

**Comment by avsej** [comment]
Turning PR to draft, as I'm going to redefine public API for replica read to follow RFC more closely

File: couchbase/api/collection.hxx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+     * @since 1.0.0
+     * @committed
+     */
+    [[nodiscard]] const std::string& name() const
+    {
+        return name_;
+    }
+
+    template<typename Handler>
+    void get_any_replica(std::string document_key, const get_any_replica_options& options, Handler&& handler) const
```
[the remark this answers is not in the document]
Comment (avsej) [reply couchbase/api/collection.hxx:83]: okay. I will fix it

**Review by davidkelly — DISMISSED** [review]
This looks good to me.   I like the sdk3 look of couchbase::api:: stuff.

**Review by davidkelly — APPROVED** [review]
LGTM!

**Comment by thejcfactor** [comment]
I like these changes!  I think what is missing would be applying the changes Brett mentioned.

**Comment by avsej** [comment]
@brett19 I want to do namespace changes in separate PR. I promise I will do it first thing after this PR will be merged

**Review by brett19 — APPROVED** [review]

**Pull request merged**